### PR TITLE
Add new module: Bustools sort

### LINKED ISSUE #946: bustools sort

### PR DESCRIPTION
**Pull request by tkhalilullah**
## PR checklist

Closes #946 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by chris-cheshire** [comment]
@Emiller88 sorry i didnt see this was self-assigned - do you want this reviewed?

**Comment by Emiller88** [comment]
@chris-cheshire I help @tkhalilullah when he wrote this module, so I was just adding myself as someone to fix any major errors. Please review it!

**Comment by chris-cheshire** [comment]
Ok thanks!

File: tests/modules/bustools/sort/test.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- name: bustools sort
+  command: nextflow run ./tests/modules/bustools/sort -entry test_bustools_sort -c tests/config/nextflow.config
+  tags:
+    - bustools
+    - bustools/sort
+  files:
+    - path: output/bustools/test.sorted.bus
```
Comment (drpatelh) [inline tests/modules/bustools/sort/test.yml:7]:
```suggestion
    - path: output/bustools/test.bus
```

File: tests/modules/bustools/sort/main.nf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { BUSTOOLS_SORT } from '../../../../modules/bustools/sort/main.nf' addParams( options: [:] )
+
+workflow test_bustools_sort {
+
+    input = [ [ id:'test' ], // meta map
+            file("https://raw.githubusercontent.com/BUStools/bustools/master/test/output.bus", checkIfExists: true) ]
```
Comment (drpatelh) [inline tests/modules/bustools/sort/main.nf:10]: We should add one of these files to nf-core/testdatasets

File: modules/bustools/sort/meta.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+name: bustools_sort
+## TODO nf-core: Add a description of the module and list keywords
```
Comment (drpatelh) [inline modules/bustools/sort/meta.yml:2]: Be great if you can sort this file out @tkhalilullah @Emiller88 and then I think it is pretty much ready to go. Can update it properly later if required.

**Comment by JoseEspinosa** [comment]
Hi @tkhalilullah, thank you for this PR! We are merging as many modules as possible now due to and impending restructuring of the entire repo. This will mean you will need to update the module to reflect these changes before it can be merged in the future. It appears like this module isn't ready to be merged so if applicable, we are converting it to draft and adding the WIP label. If this isn't the case please let us know and we will try to get the module in before the changes. Thanks again!

**Comment by matthdsm** [comment]
Hi there!

We've noticed there hasn't been much activity here. Are you still planning on working on this?
If not, you can ignore this message and we'll close your PR in about 2 weeks

Cheers
the nf-core maintainers

**Pull request closed without merging**